sdk/core/azure-core-amqp/CMakeLists.txt

```diff
@@ -192,7 +192,7 @@ az_rtti_setup(
 
 if(BUILD_TESTING)
   # define a symbol that enables some test hooks in code
-  add_compile_definitions(TESTING_BUILD)
```

Comment:
How does this change align with our new proposed rule that all per-package definitions should be defined with the package name in the definition?

Reply:
I am doing this in stages, by first fixing this issue: https://github.com/Azure/azure-sdk-for-cpp/issues/5334

This PR is purely a rename to get to an improved state than we previously were in main.

My plan is to turn this into a global definition in a subsequent PR, which is why I don't want to turn the name into per-package atm.

Reply:
Why do this in two PRs? It seems like a LOT of extra work for very little benefit.

Do we have evidence that we (or customers) are blocked by the existence of the BUILD_TESTING macro?

If we have customers that are blocked by the existence of this macro, we should do this PR, however if there is no evidence that customers are impacted, why not just do the work in a single PR?

Reply:
The rename (this PR) is already done. It was tactical. There is no extra work here.